benchmarks/devel/devel.cpp

```diff
@@ -0,0 +1,83 @@
+/****************************************************************************
+ * Copyright (c) 2017-2022 by the ArborX authors                            *
+ * All rights reserved.                                                     *
+ *                                                                          *
+ * This file is part of the ArborX library. ArborX is                       *
+ * distributed under a BSD 3-clause license. For the licensing terms see    *
+ * the LICENSE file in the top-level directory.                             *
+ *                                                                          *
+ * SPDX-License-Identifier: BSD-3-Clause                                    *
+ ****************************************************************************/
+
+#include <ArborX_DetailsUtils.hpp> // accumulate
+#include <ArborX_Version.hpp>
+
+#include <Kokkos_Core.hpp>
+
+#include <chrono>
+
+#include <benchmark/benchmark.h>
+
+template <typename ExecutionSpace>
+auto setup(ExecutionSpace const &exec_space, int n)
+{
+  using MemorySpace = typename ExecutionSpace::memory_space;
+  Kokkos::View<int *, MemorySpace> view("Benchmark::view", n);
+  return view;
+}
+
+template <typename ExecutionSpace, typename View>
+void kernel(ExecutionSpace const &exec_space, View view)
+{
+  ArborX::accumulate(exec_space, view, 0);
+}
+
+template <typename ExecutionSpace>
+void BM_benchmark(benchmark::State &state, int n)
```

Comment:
Why did you opt for this (passing `n` and wrap the call into a lambda rather than using the native way.
https://github.com/google/benchmark/blob/main/docs/user_guide.md#passing-arguments

Reply:
I find it easier to read. Also, the native way poses severe restrictions on the type (`Arg` takes in integers only in my understanding). In order to pass arbitrary arguments or type, one has to use [`BENCHMARK_CAPTURE`](https://github.com/google/benchmark/blob/main/docs/user_guide.md#passing-arbitrary-arguments-to-a-benchmark) which is convoluted.

Reply:
it also prints the value of the parameter and lets run you run the benchmark for different values of that parameter